tasks/utils.yaml

```diff
@@ -0,0 +1,33 @@
+variables:
+  - name: FLAVOR
+    default: upstream
+
+tasks:
+  - name: determine-repo
+    inputs:
+      base_repo:
+        description: The base repository to publish into
+        default: ghcr.io/defenseunicorns/packages
+      flavor:
+        description: The flavor of the package to publish
+        default: ${FLAVOR}
+      snapshot:
+        description: Whether this is a snapshot release
+        default: "false"
+    actions:
+      - description: Determine repository for the given flavor/type of release
+        cmd: |
+          repo="${{.inputs.base_repo}}"
+          # unicorn flavor = private repository
+          if [ "${{.inputs.flavor}}" = "unicorn" ]; then
+            repo="${repo}/private"
+          fi
+          repo="${repo}/uds"
```

Comment:
this should be an input passed up and out of both pull / publish

Reply:
for these people: https://github.com/defenseunicorns/leapfrogai/pkgs/container/packages%2Fleapfrogai%2Fleapfrogai-api